### ISSUE TITLE
Retrieving the Unreduced Losses

### ISSUE THREAD

**Issue by chethanjjj** [tensorflow/tensorflow#50704]
If we set the reduction in our binary cross entropy loss to tf.losses.Reduction.NONE when we compile our model (code below) is there a way to retrieve the unreduced losses when we evaluate on a sample?

```
model.compile(loss= tf.keras.losses.BinaryCrossentropy(reduction=tf.keras.losses.Reduction.NONE, from_logits=False),
              optimizer=Adam(learning_rate=1e-4))
```

I'm running into an issue where the loss in my model (with zeroed-out weights and bias - code below on how i'm doing this) is not what I'm expecting.

```
for ix, layer in enumerate(model.layers):
      weights = layer.get_weights()
      for arr in weights:
        arr[(arr > 0) | (arr < 0)] = 0
      layer.set_weights(weights)
```

So I'd like to diagnose how my model is calculating the loss by looking at output from the 3 stages of reduction: NONE, SUM, and SUM_OVER_BATCH_SIZE

**Comment by saikumarchalla** [comment]
@chethanjjj  It looks like your Issue relates to the Keras component. Please submit it to the [github.com/keras-team/keras](github.com/keras-team/keras) repository instead. As ([previously announced](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999/)), all future development of Keras is expected to happen in the [github.com/keras-team/keras](github.com/keras-team/keras) repository. If your issue lies with the TF-Core area please comment back with your explanation and we can look into it further. Thanks!
